packages/notifications/src/error.ts

```diff
@@ -31,57 +31,37 @@ export const withErrorHandling = (callback: ErrorCallback) => {
 export const withErrorIgnoring = withErrorHandling(() => {});
 
 /**
- * allows composition of NotificationsContract dependency when useful for code reuse
+ * allows composition of NotificationsProvider dependency when useful for code reuse
  * @param notifier
  */
-export const makeNotificationsContract = (
+export const makeNotificationsProvider = (
   notifier: Notifier,
-  ignoreError: IgnoreErrorDecider
-): NotificationsContract => ({
-  notify: notifier.notify,
+  shouldIgnoreError: IgnoreErrorDecider
```

Comment:
Suggest making the second argument an options object so that it's self-documenting.